File: leetcode409.go
package main

/// 409. Longest Palindrome
/*https://leetcode.com/problems/longest-palindrome/*/
func longestPalidrome(s string) int {
	count := 0
	m := make(map[byte]bool)
	for i := 0; i < len(s); i++ {
		if m[s[i]] {
			count++
			delete(m, s[i])
		} else {
			m[s[i]] = true
		}
	}
	if len(m) > 0 {
		return count*2 + 1
	}
	return count * 2
}

File: leetcode409.java
import java.util.*;
/// 409. Longest Palindrome
// author: Phan Chí Hiếu
// github: https://github.com/Phanhieu1221
/*https://leetcode.com/problems/longest-palindrome/*/
class Solution {
    public long longestPalindrome(String s) {
        Set<Character> trace = new HashSet<Character>();
        int ans = 0;
        for(int i = 0; i < s.length(); i++) {
            char x = s.charAt(i);
            if (trace.contains(x)) {
                trace.remove(x);
                ans += 2;
            } else {
                trace.add(x);
            }

        }
        return (trace.isEmpty()) ? ans : ans + 1;
    }
 
}
